core/drivers/stm32_gpio.c

```diff
@@ -321,11 +336,35 @@ static void stm32_gpio_set_direction(struct gpio_chip *chip,
 	clk_disable(bank->clock);
 }
 
-static void stm32_gpio_put_gpio(struct gpio_chip *chip __maybe_unused,
-				struct gpio *gpio)
+/* Forward reference to the PM callback handler for consumed GPIOs */
+static TEE_Result consumed_gpios_pm(enum pm_op op, unsigned int pm_hint,
+				    const struct pm_callback_handle *pm_hdl);
+
+static void stm32_gpio_put_gpio(struct gpio_chip *chip, struct gpio *gpio)
 {
+	struct stm32_gpio_bank *bank = gpio_chip_to_bank(chip);
+	struct stm32_gpio_pm_state *tstate = NULL;
+	struct stm32_gpio_pm_state *state = NULL;
+	uint32_t exceptions = 0;
+
 	assert(is_stm32_gpio_chip(chip));
-	free(gpio);
+
+	exceptions = cpu_spin_lock_xsave(&gpio_lock);
+
+	SLIST_FOREACH_SAFE(state, &consumed_gpios_head, link, tstate) {
+		if (state->gpio_pinctrl.bank == bank->bank_id &&
+		    state->gpio_pinctrl.pin == gpio->pin) {
+			SLIST_REMOVE(&consumed_gpios_head, state,
+				     stm32_gpio_pm_state, link);
+			unregister_pm_driver_cb(consumed_gpios_pm, state);
```

Comment:
`register_pm_core_service_cb()` to match line 1342.

Reply:
changed all to driver callbacks

Reply:
+1